server/job-scheduler.js

```diff
@@ -2,367 +2,424 @@ const Bree = require('bree');
-            dataflowId: messageBasedjob.dataflowId,
-            applicationId: messageBasedjob.applicationId, 
-            jobfileName: job.jobType == 'Script' ? SUBMIT_SCRIPT_JOB_FILE_NAME : SUBMIT_JOB_FILE_NAME,
             jobType: job.jobType,
-            sprayedFileScope: job.sprayedFileScope,
+            clusterId: job.cluster_id,
             sprayFileName: job.sprayFileName,
             sprayDropZone: job.sprayDropZone,
+            sprayedFileScope: job.sprayedFileScope,
+            dataflowId: messageBasedjob.dataflowId,
+            applicationId: messageBasedjob.applicationId,
+            jobfileName: job.jobType == 'Script' ? SUBMIT_SCRIPT_JOB_FILE_NAME : SUBMIT_JOB_FILE_NAME,
           });
         }
       } else {
         console.log('------------------------------------------');
-        console.error("📢 COULD NOT FIND JOB WITH NAME "+  message.jobName);
+        console.error('📢 COULD NOT FIND JOB WITH NAME ' + message.jobName);
         console.log('------------------------------------------');
       }
     } catch (err) {
       console.log(err);
     }
   }
 
- addJobToScheduler({ cron, jobfileName, sprayedFileScope, manualJob_meta, sprayFileName, sprayDropZone, applicationId, dataflowId, clusterId, metaData, jobName, contact, jobType, status, jobId, title}) {
-  try {
-        let uniqueJobName = jobName + '-' + dataflowId + '-' + jobId;
-        this.createNewBreeJob({uniqueJobName, cron, jobfileName, sprayedFileScope, manualJob_meta, sprayFileName, sprayDropZone, applicationId, dataflowId, clusterId, metaData, jobName, contact, jobType, status, jobId, title});
-        this.bree.start(uniqueJobName);
-        console.log('------------------------------------------');
-        console.log(`📢 JOB WAS SCHEDULED AS - ${uniqueJobName},  job-scheduler.js: addJobToScheduler`)
-        console.log(`📢 TOTAL SCHEDULED JOBS IN BREE (may not include dependent jobs): ${this.bree.config.jobs.length}`)
-        console.log('------------------------------------------');
-      } catch (err) {
-        console.log(err);
-      }
+  addJobToScheduler({  skipLog = false, ...jobData }) {
+    try {
+      let uniqueJobName = jobData.jobName + '-' + jobData.dataflowId + '-' + jobData.jobId;
+      this.createNewBreeJob({ uniqueJobName, ...jobData });
+      this.bree.start(uniqueJobName);
+      console.log('------------------------------------------');
+      console.log(`📢 JOB WAS SCHEDULED AS - ${uniqueJobName},  job-scheduler.js: addJobToScheduler`);
+      !skipLog && this.logBreeJobs();
+      console.log('------------------------------------------');
+      return {success: true}
+    } catch (err) {
+      console.log('-err-----------------------------------------');
+      console.dir({err}, { depth: null });
+      console.log('------------------------------------------');
+      const part2 = err.message.split(' an ')?.[1]  // error message is not user friendly, we will trim it to have everything after "an".
```

Comment:
Is this where the incorrect cron expression failure is handled? i.e. we are expecting bree to throw an error in case an incorrect cron expression is used?

Reply:
Yes, if cron is invalid Bree will throw an error, it looks like "#1 Job Tombolo-215125hljlfjaldfja failed to start, an invalid cron expression' so we will take everything after "an" if it exists, if not, we will pass whole sting